src/NumberValue.php

```diff
@@ -0,0 +1,57 @@
+<?php declare(strict_types=1);
+
+namespace GW\Value;
+
+interface NumberValue extends Value, Numberable
+{
+    // comparators
+
+    /**
+     * @return int {-1, 0, 1}
+     */
+    public function compare(Numberable $other): int;
+
+    public function equals(Numberable $other): bool;
+
+    // basic math
+
+    public function add(Numberable $other): NumberValue;
+
+    public function subtract(Numberable $other): NumberValue;
+
+    public function multiply(Numberable $other): NumberValue;
+
+    public function divide(Numberable $other): NumberValue;
```

Comment:
I think that these functions should accept plain `int` too, but it can be done in separate PR :)

Reply:
Done :+1: